From 73e91d3652bbe4cf942464f9a6c582b574d578d1 Mon Sep 17 00:00:00 2001
From: Andreas Brandner <Andreas.Brandner@cib.de>
Date: Fri, 8 Sep 2017 18:51:21 +0200
Subject: tdf#108406 offline help-images helpimg are no longer selectable as
 iconset

Change-Id: I6733ed2a04c01c288de2cd6af938a6c6ff1b4481
Reviewed-on: https://gerrit.libreoffice.org/42117
Tested-by: Jenkins <ci@libreoffice.org>
Reviewed-by: Thorsten Behrens <Thorsten.Behrens@CIB.de>
(cherry picked from commit 60adba3fb904ed66251ff39c82fe20bec151ef0b)
Reviewed-on: https://gerrit.libreoffice.org/42140
Reviewed-by: Katarina Behrens <Katarina.Behrens@cib.de>

diff --git a/cui/source/options/optgdlg.cxx b/cui/source/options/optgdlg.cxx
index 07496d9..40cbe21 100644
--- a/cui/source/options/optgdlg.cxx
+++ b/cui/source/options/optgdlg.cxx
@@ -665,6 +665,10 @@ OfaViewTabPage::OfaViewTabPage(vcl::Window* pParent, const SfxItemSet& rSet)
     StyleSettings aStyleSettings = Application::GetSettings().GetStyleSettings();
     mInstalledIconThemes = aStyleSettings.GetInstalledIconThemes();
 
+    //Removing "Helpimg" from the list, as it is not a real iconset (see tdf#108406)
+    mInstalledIconThemes.erase(std::remove_if(mInstalledIconThemes.begin(), mInstalledIconThemes.end(),
+                                [](vcl::IconThemeInfo theme) { return theme.GetDisplayName() == "Helpimg";}), mInstalledIconThemes.end());
+
     // Start with the automatically chosen icon theme
     OUString autoThemeId = aStyleSettings.GetAutomaticallyChosenIconTheme();
     const vcl::IconThemeInfo& autoIconTheme = vcl::IconThemeInfo::FindIconThemeById(mInstalledIconThemes, autoThemeId);
-- 
cgit v0.10.2

